src/main/scala/cromwell/engine/WorkflowActor.scala

```diff
@@ -1,43 +1,139 @@
 package cromwell.engine
 
-import akka.actor.Actor
+import akka.actor.{ActorRef, Props, Actor}
 import akka.actor.SupervisorStrategy.Stop
+import akka.event.LoggingReceive
+import cromwell.binding.{FullyQualifiedName, Workflow, WdlBinding, WdlValue}
 
 /**
  * This model assumes workflow actors are transient, created once per workflow instance.
  */
 object WorkflowActor {
 
-  // This is hand-waving over the actual engine-level representation of a workflow.
-  type WorkflowModel = String
+  case class Construct(binding: WdlBinding, inputs: Map[String, WdlValue])
 
-  case class Construct(model: WorkflowModel)
   case object Constructed
 
+  /**
+   * @param badInputs A `Map` of locally qualified input names to a corresponding diagnostic message.
+   */
+  case class ConstructionFailed(badInputs: Map[String, String])
+
   case object Start
+
   case object Started
 
   // Note there is no Stop message defined here, Stop is from Akka.
   case object Stopped
+
+  case class Done(outputs: Map[String, WdlValue])
+
+  case class InvalidOperation(message: String)
+
+  case class Failed(throwable: Throwable)
 }
 
-// Represents the root of a single workflow instance, not a manager of multiple
-// workflows.
+/** Represents the root of a single workflow instance, not a manager of multiple
+  * workflows.
+  */
 class WorkflowActor extends Actor {
```

Comment:
would it make sense to pass the construction stuff in as a param, i.e. a WorkflowActor is always constructed by definition?